src/main/java/bio/terra/service/snapshotbuilder/query/Concept.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.service.snapshotbuilder.query;
+
+public class Concept extends TableVariable {
```

Comment:
In general I would favor encapsulation over inheritance, unless it's unworkable due to needing to expose everything in the wrapped object via method forwarding. So here I would instead write:
```
public class Concept {
   private final TableVariable table;
```

and see how much work you can move into this class from the caller

Reply:
Having classes like `Concept` and `ConceptAncestor` extend TableVariable allows for them to be treated as TableVariables in the QueryBuilder:

```
    public Builder addTables(List<TableVariable> tables) {
      this.tables = tables;
      return this;
    }
```

and we can then do this

```
  Query createSubQuery(int conceptId) {
    // concept_relationship is primary table for the subquery
    ConceptRelationship conceptRelationship = new ConceptRelationship();
    FieldVariable descendantConceptId = conceptRelationship.concept_id_2();

    return new Query.Builder()
        .select(List.of(descendantConceptId))
        .addTables(List.of(conceptRelationship))
         ...
        .build();
  }

```